Src/StdLib/Lib/subprocess.py

```diff
@@ -1272,6 +1286,117 @@ def terminate(self):
+                           startupinfo, creationflags, shell,
+                           stdin, unused_p2cwrite,
+                           unused_c2pread, stdout,
+                           unused_errread, stderr,
+                           unused_restore_signals, unused_start_new_session):
+            """Execute program (Dotnet version)"""
+            p = Process()
+            s = p.StartInfo
+
+            if env:
+                for k, v in env.items():
+                    s.Environment[k] = v
+
+            if shell:
+                if not isinstance(args, str):
+                    args = list2cmdline(args)
+                # escape backslash and double quote
+                args = ''.join('\\' + c if c in {'\\', '"'} else c for c in args)
+                s.Arguments = '-c "{}"'.format(args)
+                s.FileName = executable or '/bin/sh'
+            else:
+                if not isinstance(args, str):
+                    s.FileName = args[0]
+                    s.Arguments = list2cmdline(args[1:])
+                else:
+                    s.FileName = args
+
+            s.RedirectStandardInput = stdin is not None
+            s.RedirectStandardOutput = stdout is not None
+            s.RedirectStandardError = stderr is not None
+            s.WorkingDirectory = cwd
+            s.UseShellExecute = False
+
+            p.Start()
+
+            self.pid = p.Id
+            self._child_created = True
+            self._handle = p
+
+            if stdin == PIPE:
+                self.stdin = open(p.StandardInput.BaseStream)
+            if stdout == PIPE:
+                self.stdout = io.BufferedReader(open(p.StandardOutput.BaseStream))
+            if stderr == PIPE:
+                self.stderr = io.BufferedReader(open(p.StandardError.BaseStream))
+
+            # dotnet an't redirect stdio to file/stream, thus has to feed from parent
+            if stdin not in (None,  DEVNULL, PIPE):
+                # assume file-like object
+                input = stdin.read()
+                with open(self._handle.StandardInput.BaseStream) as f:
+                    f.write(input)
+
+        def _internal_poll(self):
+            """Check if child process has terminated.  Returns returncode
+            attribute.
+
+            This method is called by __del__, so it can only refer to objects
+            in its local scope.
+
+            """
+            if self.returncode is None and self._handle.HasExited:
+                self.returncode = self._handle.ExitCode
+            return self.returncode
+
+        def wait(self, timeout=None, endtime=None):
+            """Wait for child process to terminate.  Returns returncode
+            attribute."""
+            if endtime is not None:
+                timeout = self._remaining_time(endtime)
+            if timeout is None:
+                self._handle.WaitForExit()
+            else:
+                self._handle.WaitForExit(timeout)
```

Comment:
I think you need to convert from float seconds to integer milliseconds?
```suggestion
                self._handle.WaitForExit(int(timeout * 1000))
```

Reply:
A lot of cases left untested, because of the unflushed `FileIO` and `BufferredWriter`. There will be more work for the `Popen` implementation once `test_subprocess` could run.